sale_coupon_multi_use/models/sale_coupon.py

```diff
@@ -0,0 +1,154 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class SaleCoupon(models.Model):
+    """Extend to implement multi use coupon rule."""
+
+    _inherit = "sale.coupon"
+
+    # Takes value from related program (coupon_multi_use field), when
+    # it is generated.
+    multi_use = fields.Boolean("Multi Use", readonly=True)
+    consumption_line_ids = fields.One2many(
+        "sale.coupon.consumption_line", "coupon_id", "Consumption Lines", readonly=True,
+    )
+    discount_fixed_amount_delta = fields.Float(
+        "Fixed Amount Delta", compute="_compute_discount_fixed_amount_delta"
+    )
+
+    def _get_discount_fixed_amount_delta(self):
+        self.ensure_one()
+        amount_total = self.program_id.discount_fixed_amount
```

Comment:
for additional saftines I would make "discount_fixed_amount" readonly if program has one used coupon

Reply:
There is constraint to not allow changing it, if there is at least one. 

Though the way Odoo implemented program and coupons relations is quite clunky. I think better way would be, if program would act just as a template and then coupons would copy values. So it would be independent from later program changes. But that would require some more improving.

For example in standard sale_coupon, when you apply some coupon discount on SO, it creates SOL with product on program. If later you change discount amount on program, it updates related product name, mentioning different amount. And then when you open that old SO again, it will show that new amount on product name as if that discount was applied, when actually it was not. So its kind of confusing to user I think.